src/rsync.rs

```diff
@@ -259,6 +260,21 @@ struct RsyncFromSnapshotWriter {
 }
 
 impl RsyncFromSnapshotWriter {
+
+    /// Creates an empty directory for the rsync out_path. Particularly needed if the snapshot
+    /// is empty since no files (and parent dirs) would be created in that case - and we want to
+    /// see an empty directory. See issue #62.
+    fn create_out_path_if_missing(&self) -> Result<()> {
+        if !self.out_path.exists() {
+            std::fs::create_dir_all(&self.out_path)
+                .with_context(|| format!("Cannot create output directory for rsync at {:?}", &self.out_path))
```

Comment:
I wondered whether `{:?}` debug mode output should be used here. I would expect Display trait impl output to be more appropriate for end user consumption, I guess if in future anyone decided the debug output should be more useful for a developer and less like something Display-able this might not be so handy. However, on deliberately causing this code to fail the error I saw was `Cannot create output directory for rsync at "/blah"` which looks fine for now I guess.

Reply:
yeah, you're right.. this is following the style used elsewhere in this project, but.. it should just use `.to_string_lossy()`.. so I will just fix that across the code base in a separate commit now and merge. It cannot affect anything - but it's just the principe of least astonishment.

Reply:
`.to_string_lossy()`, or `.display()` ?

Reply:
right `.display()` methinks.. not sure how I missed that one - I guess I just stumbled on `.to_string_lossy()` long ago and stopped looking